### PR TITLE
Constructor operations, un-split legacy IDL, respec upgrade

### LINKED ISSUE #2277: Update spec to new IDL syntax for optional dictionaries
See heycam/webidl#758

### LINKED ISSUE #2282: Update to constructor operations
See https://github.com/heycam/webidl/issues/778.

### LINKED ISSUE #2289: Overloaded operations disallowed across partial interfaces in WebIDL
[WebIDL disallows defining overloaded operations across partial interfaces](https://heycam.github.io/webidl/#idl-overloading):
> Operations must not be overloaded across interface, partial interface, interface mixin, and partial interface mixin definitions.

(this has been disallowed [for a very long time](https://github.com/heycam/webidl/commit/bae069cc35773c1d2ed0efdc1829cd7fa924a5b4))

This means our approach in splitting the legacy versions of the operations in a partial interface is invalid from a WebIDL perspective.

(this emerged as part of looking at a ReSpec upgrade for #2282 - see also https://github.com/w3c/respec/issues/2495)

### PR DESCRIPTION
**Pull request by dontcallmedom**
This pull request upgrades to latest respec so that we can use the new constructor syntax (close #2282).

Because new respec doesn't accept split IDL for overloaded operations, this pull request also fixes #2289 by importing #2291

It also incorporates #2287 and fixes #2277 - merging #2287 otherwise requires newer respec.

It fixes a few additional bugs in the process.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/webrtc-pc/pull/2292.html" title="Last updated on Sep 16, 2019, 12:43 AM UTC (c1bf737)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webrtc-pc/2292/f189de3...c1bf737.html" title="Last updated on Sep 16, 2019, 12:43 AM UTC (c1bf737)">Diff</a>

### CODE REVIEW

File: webrtc.html [hunk 1]
```diff
@@ -2373,18 +2376,26 @@ <h3>Interface Definition</h3>
   attribute EventHandler oniceconnectionstatechange;
   attribute EventHandler onicegatheringstatechange;
   attribute EventHandler onconnectionstatechange;
+
+  // Legacy Interface Extensions
+  // Supporting the methods in this section is optional.
+  // However, if these methods are supported
+  // it is mandatory to implement according to how they are specified .
```
Comment (alvestrand) [inline webrtc.html]: Suggestion: Add a link to the legacy section.
Language suggestion: "However, if these methods are supported, they MUST be implemented
as defined here". Or just delete the "However" - it is implicit in the fact that they're specified, I think.

File: webrtc.html [hunk 2]
```diff
@@ -3654,9 +3665,6 @@ <h2>Methods</h2>
       </section>
       <section>
         <h3>Legacy Interface Extensions</h3>
-          <div class="note">This section is broken out for readability. Consider
-          partial interfaces here to be part of their main counterparts, as they
-          overload existing methods.</div>
           <p>Supporting the methods in this section is optional. However,
           if these methods are supported it is mandatory to implement
           according to what is specified here.</p>
```
Comment (alvestrand) [inline webrtc.html]: Add a sentence saying that "due to WebIDL rules, the IDL syntax forthe legacy methods is given in <section>"?

Comment (dontcallmedom) [reply webrtc.html]: although I don't feel strongly about it, I don't think that's necessary - unless you know the spec as it is today by heart, you're unlikely to be surprised by having most of it in a single IDL fragment.

Comment (alvestrand) [reply webrtc.html]: The list of operations is what we usually list right under the IDL fragment. I'm looking to have the explanation of why these are listed here and not next to their IDL.

File: webrtc.html [hunk 3]
```diff
@@ -2379,8 +2379,9 @@ <h3>Interface Definition</h3>
 
   // Legacy Interface Extensions
   // Supporting the methods in this section is optional.
-  // However, if these methods are supported
-  // it is mandatory to implement according to how they are specified .
+  // if these methods are supported
```
Comment (alvestrand) [inline webrtc.html]: Looks good. Nit: Uppercase i, "in section" -> "in the section"

**Pull request merged**